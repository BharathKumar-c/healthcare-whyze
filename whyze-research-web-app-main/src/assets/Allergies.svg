<svg width="40" height="40" viewBox="0 0 40 40" fill="none" xmlns="http://www.w3.org/2000/svg">
<rect width="40" height="40" rx="20" fill="#E9FCFC"/>
<path d="M16.7748 22.2891H14.3748H13.0623H10.6998C9.4623 22.2891 8.4873 23.3016 8.4873 24.5016V25.4391C8.4873 26.6766 9.4998 27.6516 10.6998 27.6516H13.0998H14.3748H16.7748C18.0123 27.6516 18.9873 26.6391 18.9873 25.4391V24.5016C18.9873 23.3016 18.0123 22.2891 16.7748 22.2891ZM13.0623 26.3766H10.6998C10.2123 26.3766 9.7998 25.9641 9.7998 25.4766V24.5391C9.7998 24.0516 10.2123 23.6391 10.6998 23.6391H13.0998V26.3766H13.0623ZM17.6748 25.4391C17.6748 25.9266 17.2623 26.3391 16.7748 26.3391H14.3748V23.6016H16.7748C17.2623 23.6016 17.6748 24.0141 17.6748 24.5016V25.4391Z" fill="#0D5959"/>
<path d="M31.2499 18.4234L30.0499 15.9484C30.0124 15.8734 29.9749 15.8359 29.9374 15.7984V14.2609C29.9374 13.2859 29.1499 12.4609 28.1374 12.4609H24.3874C23.4124 12.4609 22.5874 13.2484 22.5874 14.2609V15.8734C22.5499 15.9109 22.5499 15.9109 22.5499 15.9484L21.3499 18.4234C21.1624 18.7609 21.0874 19.1734 21.0874 19.5859V25.9234C21.0874 27.0859 22.0249 28.0234 23.1874 28.0234H29.4124C30.5749 28.0234 31.5124 27.0859 31.5124 25.9234V19.5484C31.5499 19.1734 31.4374 18.7609 31.2499 18.4234ZM22.3624 20.5609H30.1999V23.3359H22.3624V20.5609ZM23.8999 14.2234C23.8999 13.9609 24.1249 13.7359 24.3874 13.7359H28.1749C28.4374 13.7359 28.6624 13.9609 28.6624 14.2234V15.5359H23.8999V14.2234ZM23.5249 16.8859H29.0374L30.0499 18.9859C30.0874 19.0609 30.1249 19.1359 30.1249 19.2484H22.4374C22.4749 19.1734 22.4749 19.0609 22.5124 18.9859L23.5249 16.8859ZM29.4124 26.7109H23.1874C22.7374 26.7109 22.3999 26.3359 22.3999 25.9234V24.6484H30.2374V25.9234C30.2374 26.3359 29.8624 26.7109 29.4124 26.7109Z" fill="#0D5959"/>
<path d="M13.8124 18.3875C14.6374 18.3875 15.4624 18.0875 16.0999 17.45C17.3374 16.2125 17.3374 14.15 16.0999 12.9125C14.8624 11.675 12.7999 11.675 11.5624 12.9125C10.3249 14.15 10.3249 16.2125 11.5624 17.45C12.1624 18.0875 12.9874 18.3875 13.8124 18.3875ZM15.1624 16.55C14.5999 17.1125 13.7374 17.2625 13.0249 16.925L15.5374 14.4125C15.8374 15.0875 15.7249 15.95 15.1624 16.55ZM12.4624 13.85C12.8374 13.475 13.3249 13.2875 13.8124 13.2875C14.0749 13.2875 14.3374 13.3625 14.5999 13.475L12.0874 15.9875C11.7499 15.275 11.8999 14.4125 12.4624 13.85Z" fill="#0D5959"/>
</svg>
