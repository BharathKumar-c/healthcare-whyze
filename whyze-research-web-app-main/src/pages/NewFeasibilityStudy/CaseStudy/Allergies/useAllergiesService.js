import { useCallback, useState } from 'react';
import { feasibilityStudyApiUrls } from '../../../../constant/ApiEndPoints';
import ApiUtil from '../../../../utils/ApiUtils';
import { debounce } from '../../../../utils/CommonFunction';

const useAllergiesService = () => {
  const [allergies, setAllergies] = useState([]);
  const [reactions, setReactions] = useState([]);
  const getAllAllergiesList = async (data) => {
    try {
      await ApiUtil.getData(
        `${feasibilityStudyApiUrls.allergiesListSearchUrl}?value=${data}`,
      ).then((res) => {
        setAllergies(
          res.data.result.map((ele, index) => ({
            value: ele.Name,
            id: index,
          })),
        );
      });
    } catch (error) {
      console.log(error);
    }
  };
  const getAllReactionList = async (data) => {
    try {
      await ApiUtil.getData(
        `${feasibilityStudyApiUrls.reactionListSearchUrl}?value=${data}`,
      ).then((res) => {
        setReactions(
          res.data.result.map((ele, index) => ({
            value: ele.ReactionType,
            id: index,
          })),
        );
      });
    } catch (error) {
      console.log(error);
    }
  };
  const optimusValue = useCallback(debounce(getAllAllergiesList, 300), []);
  const reactionDebounce = useCallback(debounce(getAllReactionList, 300), []);
  return {
    allergies,
    reactions,
    optimusValue,
    reactionDebounce,
  };
};

export default useAllergiesService;
