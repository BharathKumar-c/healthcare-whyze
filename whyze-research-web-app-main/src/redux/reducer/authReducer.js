import { createSlice } from '@reduxjs/toolkit';
import jwt_decode from 'jwt-decode';

const initialState = {
  isLoggedIn: loginCheck(),
  accessToken: null,
  refreshToken: null,
  isLoading: false,
  isDisableForgot: false,
  message: '',
  userDetails: {},
  profileCount: 0,
};

function loginCheck() {
  const credentials = JSON.parse(localStorage.getItem('credentials'));
  if (credentials?.access_token) {
    const { exp } = jwt_decode(credentials.access_token);
    if (Date.now() <= exp * 1000) {
      return true;
    }
  }
  return false;
}

const authSlice = createSlice({
  name: 'authAction',
  initialState: initialState,
  reducers: {
    setLoginStatus: (state, { payload }) => {
      state.isLoggedIn = payload;
    },
    setAccessToken: (state, { payload }) => {
      state.accessToken = payload;
    },
    setRefreshToken: (state, { payload }) => {
      state.refreshToken = payload;
    },
    setLoading: (state, { payload }) => {
      state.isLoading = payload;
    },
    setDisableForgot: (state, { payload }) => {
      state.isDisableForgot = payload;
    },
    setResetMessage: (state, { payload }) => {
      state.message = payload;
    },
    setUserDetails: (state, { payload }) => {
      state.userDetails = payload;
    },
    setProfileCount: (state) => {
      state.profileCount = state.profileCount + 1;
    },
  },
});

export const {
  setLoginStatus,
  setAccessToken,
  setRefreshToken,
  setLoading,
  setDisableForgot,
  setResetMessage,
  setUserDetails,
  setProfileCount,
} = authSlice.actions;

export default authSlice.reducer;
