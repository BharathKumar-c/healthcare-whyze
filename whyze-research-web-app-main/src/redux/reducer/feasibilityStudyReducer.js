import { createSlice } from '@reduxjs/toolkit';
import { v4 as uuid } from 'uuid';

export const initialState = {
  segments: [
    {
      name: 'Segment-1',
      isDone: false,
      activeTab: {
        name: 'feasibility',
        index: 0,
      },
      tabs: [
        {
          name: 'feasibility',
          title: 'Feasibility',
          isActive: true,
          position: 'left',
        },
      ],
      feasibilityStudy: [
        {
          name: 'Indication',
          case: 'Indication',
          condition: 'Inclusion',
          caseName: '',
          child: [],
        },
        {
          name: 'Demographics',
          case: 'Demographics',
          condition: 'Inclusion',
          caseName: '',
          child: [
            {
              id: uuid().slice(0, 8),
              condition: 'Gender',
              field: 'gender',
              gender: '',
            },
            {
              id: uuid().slice(0, 8),
              condition: 'Age range',
              field: 'age',
              ageRange: {
                min: 0,
                max: 0,
              },
            },
          ],
        },
      ],
      country: [],
      location: [],
      selectedCountries: [],
      totalConnectedNotConnected: {},
      sites: [],
      sitesContacted: [],
      sitesInitiated: [],
      preferredSites: [],
      patientCount: {
        feasibility: 0,
        location: 0,
      },
      siteCount: {
        feasibility: 0,
        location: 0,
      },
    },
  ],
  cardValue: {
    case: 'Indication',
    condition: '',
    index: 0,
  },
  activeSegment: {
    name: '',
    index: 0,
  },
};

const feasibilityStudySlice = createSlice({
  name: 'feasibilityStudyAction',
  initialState: initialState,
  reducers: {
    setCardDetails: (state, { payload }) => {
      state.cardValue = payload;
    },
    setSegments: (state, { payload }) => {
      state.segments = payload;
    },
    setActiveSegment: (state, { payload }) => {
      state.activeSegment = payload;
    },
    resetSegments: (state) => {
      state.segments = initialState.segments;
      state.cardValue = {
        case: 'Indication',
        condition: '',
        index: 0,
      };
      state.activeSegment = {
        name: '',
        index: 0,
      };
    },
  },
});

export const { setCardDetails, setSegments, setActiveSegment, resetSegments } =
  feasibilityStudySlice.actions;

export default feasibilityStudySlice.reducer;
