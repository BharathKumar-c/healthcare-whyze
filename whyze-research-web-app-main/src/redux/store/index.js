import { configureStore } from '@reduxjs/toolkit';
import reducer from '../reducer/combinedReducer';

const store = configureStore({
  reducer: reducer,

  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export default store;
