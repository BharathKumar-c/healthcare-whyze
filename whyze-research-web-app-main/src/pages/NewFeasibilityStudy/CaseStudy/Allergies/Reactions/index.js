import React, { useEffect, useState } from 'react';
import {
  AutoCompleteComponent,
  CustomSelect,
  Text,
} from '../../../../../components/basic';
import { ConditionContainer } from '../../../../../components/shared';
import PropTypes from 'prop-types';
import './Reactions.scss';
import { Image, Row, Space, Tag } from 'antd';
import {
  autocompleteDefaultLabel,
  autocompleteEmptyLabel,
  conditionOptions,
} from '../../../../../constant';
import DragIcon from '../../../../../assets/DragIcon.svg';
import AddIcon from '../../../../../assets/AddIcon.svg';
import CloseIcon from '../../../../../assets/CloseIcon.svg';
import { useSelector, useDispatch } from 'react-redux';
import { setSegments } from '../../../../../redux/reducer/feasibilityStudyReducer';
import {
  addChildInCaseStudy,
  removeChildInCaseStudy,
  updateCaseStudy,
  updateConditionChanges,
} from '../../../../../components/shared/AddCondition';
import { v4 as uuid } from 'uuid';
import { Vector } from '../../../../../assets';
import Allergies from '..';
import useNewFeasibilityStudyService from '../../../useNewFeasibilityStudyService';
import { totalPatientFormatter } from '../../../../../utils/calculations';

const Reactions = ({
  reactions,
  totalpatient,
  caseDetails,
  parentIndex,
  parentId,
  isChild,
  operationValue,
  handleAddCondition,
  handleChangeCaseName,
  getPopupContainer,
}) => {
  const { segments, activeSegment } = useSelector(
    (state) => state.feasibilityStudyReducer,
  );
  const [isGetCount, setIsGetCount] = useState(false);

  const dispatch = useDispatch();
  const { optimusValue } = useNewFeasibilityStudyService();

  const handleAddNewCondition = (id) => {
    const study = JSON.parse(JSON.stringify(segments));
    addChildInCaseStudy(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      id,
      {
        id: uuid().slice(0, 8),
        condition: '',
        operation: 'And',
        child: [],
      },
    );
    dispatch(setSegments(study));
  };

  const handleSelectRelation = (value, id) => {
    const study = JSON.parse(JSON.stringify(segments));
    const newValue = conditionOptions.find((a) => a.key === value).label;
    updateCaseStudy(
      study[activeSegment.index]?.feasibilityStudy[parentIndex].child,
      id,
      'operation',
      newValue,
    );
    optimusValue({
      id,
      data: study[activeSegment.index]?.feasibilityStudy,
    });
    dispatch(setSegments(study));
  };

  const handleRemoveCondition = (id) => {
    const study = JSON.parse(JSON.stringify(segments));
    removeChildInCaseStudy(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      parentId,
      id,
    );
    dispatch(setSegments(study));
  };

  const handleChangeSymptom = (value) => {
    const study = JSON.parse(JSON.stringify(segments));
    updateConditionChanges(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      caseDetails?.id,
      'reactions',
      {
        symptom: value,
      },
    );
    dispatch(setSegments(study));
  };

  const handleSelectSymptom = (value) => {
    const study = JSON.parse(JSON.stringify(segments));
    updateConditionChanges(
      study[activeSegment.index].feasibilityStudy[parentIndex].child,
      caseDetails?.id,
      'reactions',
      {
        symptom: value,
      },
    );
    setIsGetCount(true);

    dispatch(setSegments(study));
  };

  useEffect(() => {
    if (isGetCount) {
      optimusValue({
        id: caseDetails.id,
        data: segments[activeSegment.index]?.feasibilityStudy,
      });
      setIsGetCount(false);
    }
  }, [isGetCount]);

  return (
    <>
      <ConditionContainer
        className={'condition-container'}
        isActive
        isVertical
        isChild={isChild}
      >
        <Space direction="vertical" className="container">
          <div className="container-alignment" id="container-alignment">
            <div className="drag-drop-image">
              <Image src={DragIcon} width={10} height={16} preview={false} />
            </div>
            <div className="add-image">
              <Image
                className="image"
                src={AddIcon}
                width={12}
                height={12}
                preview={false}
                onClick={() => handleAddNewCondition(caseDetails.id)}
              />
            </div>
            <div className="condition-container-before-select" />
            <CustomSelect
              className="condition-container-custom-select"
              options={conditionOptions}
              onSelect={(value) => handleSelectRelation(value, caseDetails.id)}
              defaultValue={operationValue}
              getPopupContainer={getPopupContainer}
            />
            <div className="condition-container-after-select" />
            <Text className="indication-text" text="Allergies at least" />
            <AutoCompleteComponent
              options={
                caseDetails?.reactions?.symptom?.length >= 3
                  ? reactions.length > 0
                    ? reactions
                    : autocompleteEmptyLabel
                  : autocompleteDefaultLabel
              }
              getPopupContainer={() =>
                document.getElementById('container-alignment')
              }
              placeholder={'Symptoms / Reaction'}
              label={'Symptoms / Reaction'}
              width={'420px'}
              value={caseDetails?.reactions?.symptom}
              handleSearch={handleChangeSymptom}
              handleSelect={handleSelectSymptom}
            />
            <div className="close-icon">
              <Image
                src={CloseIcon}
                className="image"
                width={20}
                height={20}
                preview={false}
                onClick={() => handleRemoveCondition(caseDetails.id)}
              />
            </div>
          </div>
          <Row className="indication-no-patient-tag-row">
            <Tag className="indication-no-patient-tag">
              <Vector />
              {totalPatientFormatter(totalpatient || 0)}
            </Tag>
          </Row>
        </Space>
        {caseDetails?.child?.length > 0 && (
          <Allergies
            isChild
            parentIndex={parentIndex}
            caseDetails={caseDetails}
            handleAddCondition={handleAddCondition}
            handleChangeCaseName={handleChangeCaseName}
          />
        )}
      </ConditionContainer>
    </>
  );
};
Reactions.propTypes = {
  reactions: PropTypes.array,
  index: PropTypes.number,
  totalpatient: PropTypes.string,
  caseDetails: PropTypes.object,
  parentId: PropTypes.string,
  operationValue: PropTypes.string,
  parentIndex: PropTypes.number,
  isChild: PropTypes.bool,
  getPopupContainer: PropTypes.string,
  handleAddCondition: PropTypes.func,
  handleChangeCaseName: PropTypes.func,
};

Reactions.defaultProps = {
  reactions: [],
  index: 0,
  totalpatient: '',
  caseDetails: null,
  parentId: '',
  operationValue: '',
  getPopupContainer: '',
  parentIndex: 0,
  isChild: false,
  handleAddCondition: () => {},
  handleChangeCaseName: () => {},
};
export default Reactions;
