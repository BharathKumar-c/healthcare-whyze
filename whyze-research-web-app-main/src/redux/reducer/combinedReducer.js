import { combineReducers } from 'redux';
import feasibilityStudy from './feasibilityStudyReducer';
import project from './projectsReducer';
import auth from './authReducer';

export default combineReducers({
  feasibilityStudyReducer: feasibilityStudy,
  authReducer: auth,
  projectReducer: project,
});
