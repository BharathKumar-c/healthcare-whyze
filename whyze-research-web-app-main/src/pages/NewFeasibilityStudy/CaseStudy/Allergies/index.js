import React, { useEffect, useState } from 'react';
import { Image, AutoCompleteComponent } from '../../../../components/basic';
import AddIcon from '../../../../assets/AddIcon.svg';
import PropTypes from 'prop-types';
import {
  AllergiesOptions,
  Allergiesarray,
  autocompleteDefaultLabel,
  autocompleteEmptyLabel,
  Symptomarray,
} from '../../../../constant';

import AddCondition from '../../../../components/shared/AddCondition';
import Reactions from './Reactions';
import useNewFeasibilityStudyService from '../../useNewFeasibilityStudyService';
import { useSelector } from 'react-redux';
import { allergiesConstants } from '../../../../constant/ConstantTexts';

const Allergies = (props) => {
  const [isError, setIsError] = useState(false);
  const [isGetCount, setIsGetCount] = useState(false);
  const { segments, activeSegment } = useSelector(
    (state) => state.feasibilityStudyReducer,
  );
  const { optimusValue } = useNewFeasibilityStudyService();

  const handleError = () => {
    setIsError(true);
  };
  const handleChangeCaseName = (e) => {
    props.handleChangeCaseName(e, props.parentIndex);
    setIsError(false);
  };

  const handleSelectCaseName = (e) => {
    props.handleChangeCaseName(e, props.parentIndex);
    setIsError(false);
    setIsGetCount(true);
  };

  useEffect(() => {
    if (isGetCount) {
      optimusValue({
        name: props?.caseDetails?.name,
        data: segments[activeSegment.index]?.feasibilityStudy,
      });
      setIsGetCount(false);
    }
  }, [isGetCount]);

  return (
    <div className="indication" id="indication">
      {!props?.isChild && (
        <div>
          <AutoCompleteComponent
            options={
              props?.caseDetails?.caseName?.length >= 3
                ? Allergiesarray.length > 0
                  ? Allergiesarray
                  : autocompleteEmptyLabel
                : autocompleteDefaultLabel
            }
            getPopupContainer={() => document.getElementById('indication')}
            value={props?.caseDetails?.caseName}
            placeholder={allergiesConstants.allergiesPlaceholder}
            label={allergiesConstants.allergiesLabel}
            handleSearch={handleChangeCaseName}
            handleSelect={handleSelectCaseName}
            errorText={allergiesConstants.allergiesErrorText}
            isError={isError}
            width={'320px'}
          />
          <div style={{ height: '30px' }}>
            <Image
              style={{ cursor: 'pointer' }}
              preview={false}
              width={12}
              alt="allergies-add-icon"
              height={12}
              src={AddIcon}
              onClick={() =>
                props?.caseDetails?.caseName !== ''
                  ? props.handleAddCondition(props.parentIndex)
                  : handleError()
              }
            />
          </div>
        </div>
      )}

      {props.caseDetails?.child?.map((c, index) => (
        <div
          key={`${index}_${c.condition}_${c.operation}`}
          id="allergies-addcondition"
        >
          {props.isChild ? (
            <div
              className={
                c.condition === '' ? 'vertical-line-condition' : 'vertical-line'
              }
            />
          ) : (
            <div
              style={{
                borderLeft: `2px ${
                  c.condition === '' ? 'dashed' : 'solid'
                } #D3D6D9`,
                height: '32px',
              }}
            />
          )}
          {c.condition === '' ? (
            <AddCondition
              caseDetails={c}
              isChild={props.isChild}
              parentId={props.caseDetails.id}
              parentIndex={props.parentIndex}
              value={c.operation}
              addConditionOptions={AllergiesOptions}
              handleAddCondition={props.handleAddCondition}
              handleChangeCaseName={props.handleChangeCaseName}
              conditionName="allergies"
              getPopupContainer={'allergies-addcondition'}
            />
          ) : c.condition === 'reactions' ? (
            <Reactions
              reactions={Symptomarray}
              caseDetails={c}
              isChild={props.isChild}
              parentIndex={props.parentIndex}
              parentId={props.caseDetails.id}
              operationValue={c.operation}
              totalpatient={c?.count}
              handleAddCondition={props.handleAddCondition}
              handleChangeCaseName={props.handleChangeCaseName}
              getPopupContainer="indication"
            />
          ) : (
            ''
          )}
        </div>
      ))}
    </div>
  );
};

Allergies.propTypes = {
  caseDetails: PropTypes.array,
  parentIndex: PropTypes.number,
  isChild: PropTypes.bool,
  handleAddCondition: PropTypes.func,
  handleChangeCaseName: PropTypes.func,
};

Allergies.defaultProps = {
  caseDetails: [],
  parentIndex: 0,
  isChild: false,
  handleAddCondition: () => {},
  handleChangeCaseName: () => {},
};
export default Allergies;
