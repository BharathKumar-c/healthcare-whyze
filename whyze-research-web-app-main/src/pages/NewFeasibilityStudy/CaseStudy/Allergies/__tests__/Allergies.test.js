import { act, fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../../../../redux/store';
import React from 'react';
import Allergies from '..';
import { setSegments } from '../../../../../redux/reducer/feasibilityStudyReducer';
function renderWithContext(element) {
  const { container, getByText } = render(
    <Provider store={store}>{element}</Provider>,
  );
  return { container, store, getByText };
}
const caseDetails = {
  name: 'Allergies',
  case: 'Allergies',
  condition: 'Inclusion',
  caseName: '',
  child: [],
};
describe('create tests for alleriges', () => {
  test('create snapshots for allergies', () => {
    const { container } = renderWithContext(<Allergies />);
    act(() => {
      const { segments, activeSegment } = JSON.parse(
        JSON.stringify(store.getState().feasibilityStudyReducer),
      );
      segments[activeSegment.index].feasibilityStudy.push(caseDetails);
      store.dispatch(setSegments(segments));
    });
    expect(container).toMatchSnapshot();
  });
  test('Should render the input field correctly', () => {
    renderWithContext(<Allergies />);
    const inputField = screen.getByRole('combobox');

    expect(inputField).toBeInTheDocument();
    expect(inputField.value).toMatch('');

    fireEvent.change(inputField, {
      target: { value: 'Sasm' },
    });
    expect(inputField.value).toMatch('Sasm');
  });
  test('Should render the Add icon correctly', () => {
    renderWithContext(<Allergies />);

    const addIcon = screen.getByAltText('allergies-add-icon');
    expect(addIcon).toBeInTheDocument();
  });
  test('Should render when click add icon with caseName correctly', async () => {
    const caseDetailsWithCaseName = {
      name: 'Allergies',
      case: 'Allergies',
      condition: 'Inclusion',
      caseName: 'Sesame Seeds',
      child: [],
    };

    const handleAddCondition = jest.fn();
    renderWithContext(
      <Allergies
        caseDetails={caseDetailsWithCaseName}
        parentIndex={1}
        handleAddCondition={handleAddCondition}
      />,
    );

    const addIcon = screen.getByAltText('allergies-add-icon');
    fireEvent.click(addIcon);

    expect(handleAddCondition).toBeCalledTimes(1);
  });
  test('Should render  component when allergies has reactions child component', () => {
    const caseDetailsWithCaseName = {
      name: 'Allergies',
      case: 'Allergies',
      condition: 'Inclusion',
      caseName: 'Sesame Seeds',
      child: [
        {
          id: 'b53a26c3',
          condition: 'reactions',
          operation: 'And',
          reactions: {
            symptom: 'Sneezing',
          },
        },
      ],
    };
    const { container } = renderWithContext(
      <Allergies caseDetails={caseDetailsWithCaseName} isChild />,
    );
    expect(container).toBeInTheDocument();
  });
  test('Should render reaction component when allergies has empty child component', () => {
    const caseDetailsWithCaseName = {
      name: 'Allergies',
      case: 'Allergies',
      condition: 'Inclusion',
      caseName: 'Sesame Seeds',
      child: [
        {
          id: 'b53a26c3',
          condition: '',
          operation: 'And',
          child: [],
        },
      ],
    };
    const { container } = renderWithContext(
      <Allergies caseDetails={caseDetailsWithCaseName} isChild />,
    );
    expect(container).toBeInTheDocument();
  });
});
