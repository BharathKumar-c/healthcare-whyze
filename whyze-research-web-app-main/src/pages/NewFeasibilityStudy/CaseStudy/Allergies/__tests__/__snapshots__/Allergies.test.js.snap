// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create tests for alleriges create snapshots for allergies 1`] = `
<div>
  <div
    class="indication"
  >
    <div>
      <div
        class="autoComplete-wrapper"
      >
        <span
          class="ant-typography input-lable"
        >
          Allergies condition
        </span>
        <div
          class="ant-select ant-select-auto-complete autoComplete-wrapper-input ant-select-single ant-select-show-search"
          style="width: 320px;"
        >
          <div
            class="ant-select-selector"
          >
            <span
              class="ant-select-selection-search"
            >
              <input
                aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                aria-autocomplete="list"
                aria-controls="rc_select_TEST_OR_SSR_list"
                aria-haspopup="listbox"
                aria-owns="rc_select_TEST_OR_SSR_list"
                autocomplete="off"
                class="ant-select-selection-search-input"
                id="rc_select_TEST_OR_SSR"
                role="combobox"
                type="search"
                value=""
              />
            </span>
            <span
              class="ant-select-selection-placeholder"
            >
              Allergies condition
            </span>
          </div>
        </div>
      </div>
      <div
        style="height: 30px;"
      >
        <div
          class="ant-image"
          style="width: 11px; height: 11px;"
        >
          <img
            alt="allergies-add-icon"
            class="ant-image-img"
            height="11"
            src="AddIcon.svg"
            style="height: 11px; cursor: pointer;"
            width="11"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
